app/validators/placeholders/placeholder_validator.py

```diff
@@ -108,6 +108,46 @@ def validate_option_label_from_value_placeholder(self, answer_id):
                 identifier=answer_id,
             )
 
+    def validate_answer_type_for_transform(
+        self, argument, argument_name, transform_type
+    ):
+        if (
+            transform_type in ["format_unit", "format_percentage"]
+            and argument_name == "value"
+            and argument.get("source") == "answers"
+        ):
+            answer_id = argument.get("identifier")
+            answer_type = self.questionnaire_schema.answers_with_context[answer_id][
+                "answer"
+            ]["type"]
+            if answer_type.lower() not in transform_type:
+                self.add_error(
+                    error_messages.ANSWER_TYPE_FOR_TRANSFORM_TYPE_INVALID.format(
+                        transform=transform_type, answer_type=answer_type
+                    ),
+                    identifier=answer_id,
+                )
+
+    def validate_answer_and_transform_unit_match(self, arguments, transform_type):
```

Comment:
I know we haven't done it for all, but given we dont have type checks for this repo, worth enforcing keyword arguments for all these methods to avoid passing args in the wrong order. So using `*, ` etc.

Reply:
Added keyword arguments when invoking, I hope that is what you meant?

Reply:
Ok added `*,` so only keyword args accepted, thanks for the read!